### PR TITLE
Revert OTel top-level spans identification changes

### PR DESCRIPTION
**Pull request by liustanley**
### What does this PR do?

Reverts the changes from the following PRs:
- https://github.com/DataDog/datadog-agent/pull/24156
- https://github.com/DataDog/datadog-agent/pull/23945
- https://github.com/DataDog/datadog-agent/pull/22163

### Motivation

The OTel top-level spans identification changes have been delayed until we have a new policy for releasing breaking changes for OTel customers.

### CODE REVIEW

**Review by dustmop — APPROVED** [review]
LGTM for agent-shared-components owned files

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -1553,8 +1553,8 @@ api_key:
   ## @env DD_APM_COMPUTE_STATS_BY_SPAN_KIND - bool - default: false
   ## [BETA] Enables an additional stats computation check on spans to see they have an eligible `span.kind` (server, consumer, client, producer).
   ## If enabled, a span with an eligible `span.kind` will have stats computed. If disabled, only top-level and measured spans will have stats computed.
-  ## NOTE: Stats are computed by span kind for OTel traces by default.
-  ## If you are sending OTel traces and do not want stats computed by span kind, you need to disable this flag and enable the "disable_otlp_compute_top_level_by_span_kind" APM feature.
+  ## NOTE: For stats computed from OTel traces, only top-level spans are considered when this option is off.
+  ## If you are sending OTel traces and want stats on non-top-level spans, this flag will need to be enabled.
```
Comment (brett0000FF) [inline pkg/config/config_template.yaml:1557]:
```suggestion
  ## If you are sending OTel traces and want stats on non-top-level spans, this flag needs to be enabled.
```

**Review by brett0000FF — APPROVED** [review]
Just a small suggestion from Docs team to tweak the tense.

**Comment by liustanley** [comment]
Closing in favor of https://github.com/DataDog/datadog-agent/pull/24232

**Pull request closed without merging**